### ISSUE TITLE
False Positives

### ISSUE THREAD

**Issue by OneEye134** [tboy1337/Blinter#31]
Here's the code I ran through the linter:

```
@echo off

setlocal EnableDelayedExpansion

call :add 15 27 result

echo Result: !result!

exit /b


REM :add
REM Adds two numbers.
REM %1 = First number
REM %2 = Second number
REM %3 = Output variable
:add
set /a "%~3=%~1 + %~2"
exit /b
```

and the linter said line 18 had a E019 error, even though the code worked normal.
 
Here is the full linter output:

```
C:\Users\Chidee\Desktop\Code\Batch++\helper>blinter class.bat
Blinter v1.1.3 - Batch File Linter


 Batch File Analysis: class.bat
==================================
Scripts Analyzed:
  1. class.bat


DETAILED ISSUES:
----------------

ERROR LEVEL ISSUES:
===================

Line 18: Percent-tilde on non-parameter variable (E019)
- Explanation: Percent-tilde syntax only works with command-line parameters (%1-%9) and FOR loop variables
- Recommendation: Use percent-tilde only with %1-%9 parameters or FOR loop variables like %%i
- Context: Percent-tilde syntax used on invalid parameter: 3=


WARNING LEVEL ISSUES:
=====================

Line 7: Potentially undefined variable reference (E006)
- Explanation: Script references variables that were never set in this script. This may be intentional if using environment variables, but could cause runtime errors if the variable is not set by parent process or system
- Recommendation: If this is an environment variable, this warning can be ignored. Otherwise, define the variable using SET before referencing it, or add IF DEFINED checks to handle undefined cases
- Context: Variable 'RESULT' is used but never defined

Line 18: Errorlevel handling difference between .bat/.cmd (W028)
- Explanation: Commands like APPEND, DPATH, FTYPE, SET, PATH, ASSOC handle errorlevel differently in .bat vs .cmd files
- Recommendation: Use .cmd extension for consistent errorlevel behavior with these commands
- Context: Command 'set' handles errorlevel differently in .bat vs .cmd files


STYLE LEVEL ISSUES:
===================

Line 1: BAT extension used instead of CMD for newer Windows (S007)
- Explanation: The .cmd file extension is recommended over .bat for Windows NT and newer versions (Windows 2000+). CMD files support additional features and have better error handling in newer Windows environments
- Recommendation: Consider renaming .bat files to .cmd for scripts intended for Windows 2000 and newer versions. CMD files provide better compatibility with modern Windows features and improved error reporting
- Context: Consider using .cmd extension instead of .bat for scripts targeting Windows 2000 and newer


PERFORMANCE LEVEL ISSUES:
=========================

Line 9, 19: Missing ENDLOCAL before exit (P006)
- Explanation: SETLOCAL should be paired with ENDLOCAL before every exit point
- Recommendation: Add ENDLOCAL before all EXIT statements when SETLOCAL is used
- Context: EXIT with active SETLOCAL should be preceded by ENDLOCAL


SEVERITY BREAKDOWN:
====================

Error: 1 issue
  Critical issues that will cause script failure or incorrect behavior.

Warning: 2 issues
  Issues that may cause problems or unexpected behavior.

Style: 1 issue
  Code style and formatting issues that affect readability.

Performance: 2 issues
  Performance issues and optimization opportunities.

WARNING  Found 1 critical issue (errors or security) that must be fixed.

C:\Users\Chidee\Desktop\Code\Batch++\helper>
```

**Comment by tboy1337** [comment]
Hi, thanks for reporting the issue.  It should now be fixed in v1.1.4 https://github.com/tboy1337/Blinter/commit/bfc00c0e6bb83facc5d1eb464caf34d8b2d3b34c
